coral-incremental/src/test/java/com/linkedin/coral/incremental/RelNodeCostEstimatorTest.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2024 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.incremental;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.calcite.rel.RelNode;
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.linkedin.coral.incremental.TestUtils.*;
+import static org.testng.Assert.*;
+
+
+public class RelNodeCostEstimatorTest {
+  private HiveConf conf;
+
+  private RelNodeCostEstimator estimator;
+
+  static final String TEST_JSON_FILE_DIR = "src/test/resources/";
+
+  @BeforeClass
+  public void beforeClass() throws HiveException, MetaException, IOException {
+    conf = TestUtils.loadResourceHiveConf();
+    estimator = new RelNodeCostEstimator(2.0, 1.0);
+    TestUtils.initializeViews(conf);
+  }
+
+  @AfterTest
+  public void afterClass() throws IOException {
+    FileUtils.deleteDirectory(new File(conf.get(CORAL_INCREMENTAL_TEST_DIR)));
+  }
+
+  public Map<String, Double> fakeStatData() {
+    Map<String, Double> stat = new HashMap<>();
+    stat.put("hive.test.bar1", 80.0);
+    stat.put("hive.test.bar2", 20.0);
+    stat.put("hive.test.bar1_prev", 40.0);
+    stat.put("hive.test.bar2_prev", 10.0);
+    stat.put("hive.test.bar1_delta", 60.0);
+    stat.put("hive.test.bar2_delta", 10.0);
+    return stat;
+  }
+
+  @Test
+  public void testSimpleSelectAll() throws IOException {
+    String sql = "SELECT * FROM test.bar1";
+    RelNode relNode = hiveToRelConverter.convertSql(sql);
+    estimator.loadStatistic(TEST_JSON_FILE_DIR + "statistic.json");
+    assertEquals(estimator.getCost(relNode), 300.0);
+    estimator.setStat(fakeStatData());
+    assertEquals(estimator.getCost(relNode), 240.0);
+  }
+
+  @Test
+  public void testSimpleJoin() throws IOException {
+    String sql = "SELECT * FROM test.bar1 JOIN test.bar2 ON test.bar1.x = test.bar2.x";
```

Comment:
The idea is not to get rid of incremental computation from the cost estimator testing.

We want to test that given a correct list of  RelNodes representing the batch, partially incremental and fully incrementally of a 2-table join, does this estimator compute the right cost for each and eventual return the cheapest plan?

So you can hard-code the correct batch, partially incremental and fully incrementally queries and feed them into `hiveToRelConverter.convertSql`, then pass the resulting RelNodes to your cost estimator.

Reply:
Yeah, I see. 
I used your method before but failed. It's because `hiveToRelConverter.convertSql` couldn't handle the sql for 'table_prev' and 'table_delta', (didn't register the table_prev like schema before), thus can't convert it to RelNode. 
I think for this PR, not using incremental computation is OK because the incremental format is just a union with three joins ? Could discuss this more. :)

Reply:
Will test table with 'prev' and 'delta' format once plan generation part is integrated.
Now only test raw format with multi-format query.